exchange/exchangeUtils.js

```diff
@@ -0,0 +1,58 @@
+// generic low level reusuable utils for interacting with exchanges.
+
+const retry = require('retry');
+const errors = require('./exchangeErrors');
+
+const retryInstance = (options, checkFn, callback) => {
```

Comment:
You could use a default value for the options:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/Default_parameters